x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,199 @@
+                      <EuiIconTip
+                        position="right"
+                        content={i18n.translate('xpack.lens.indexPattern.ranges.maxIntervalsHelp', {
+                          defaultMessage:
+                            "Intervals will be selected automatically based on the available data. The maximum number of bars can never be greater than the Advanced Setting's {histogramMaxBars}",
+                          values: { histogramMaxBars: UI_SETTINGS.HISTOGRAM_MAX_BARS },
+                        })}
+                        type="questionInCircle"
+                      />
+                    </>
+                  }
+                />
+              ) : (
+                <EuiFieldNumber
+                  data-test-subj="lns-indexPattern-range-interval-field"
+                  value={interval}
+                  onChange={({ target }) => onIntervalChange(Number(target.value))}
+                  prepend={
+                    <>
+                      <EuiText size="s">
+                        {i18n.translate('xpack.lens.indexPattern.ranges.min', {
+                          defaultMessage: 'Min Interval',
+                        })}
+                      </EuiText>{' '}
+                      <EuiIconTip
+                        position="right"
+                        content={i18n.translate('xpack.lens.indexPattern.ranges.minIntervalsHelp', {
+                          defaultMessage:
+                            "Interval will be automatically scaled in the event that the provided value creates more buckets than specified by Advanced Setting's {histogramMaxBars}",
+                          values: { histogramMaxBars: UI_SETTINGS.HISTOGRAM_MAX_BARS },
+                        })}
+                        type="questionInCircle"
+                      />
+                    </>
+                  }
+                  min={0}
+                />
+              )}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiButtonEmpty iconType="controlsHorizontal" onClick={() => onToggleEditor()}>
+            {i18n.translate('xpack.lens.indexPattern.ranges.customIntervalsToggle', {
+              defaultMessage: 'Create custom intervals',
+            })}
+          </EuiButtonEmpty>
+        </>
+      </EuiFormRow>
+    </>
+  );
+};
+
+export const RangeEditor = ({
+  onAutoIntervalToggle,
+  setParam,
+  params,
+  maxHistogramBars,
+  onChangeMode,
+  rangeFormatter,
+}: {
+  params: RangeColumnParams;
+  maxHistogramBars: number;
+  setParam: UpdateParamsFnType;
+  onAutoIntervalToggle: (enabled: boolean) => void;
+  onChangeMode: (mode: MODES_TYPES) => void;
+  rangeFormatter: IFieldFormat;
+}) => {
+  const [isAdvancedEditor, toggleAdvancedEditor] = useState(params.type === MODES.Range);
+  const isAutoIntervalEnabled = isAutoInterval(params.interval);
+  const numericIntervalValue: number = isAutoIntervalEnabled ? 0 : (params.interval as number);
```

Comment:
This should have an empty state, because we shouldn't allow `0` as a valid minimum interval. It gets scaled up into auto intervals, but this is confusing. It also makes it impossible to type a number that doesn't start with 0 into the number input, so all my numbers look like `0100` instead of `100`. For both these reasons it should have an empty state.

Reply:
I think we're talking about two distinct issues here: 
* there's an issue when changing values passing through an empty state. This is ok to solve.
* there's another issue when transitioning from `Auto` intervals to manual intervals where empty state should be represented somehow.

In Visualize there's no "reactive" behaviour so the empty state is currently not valid when transitioning to manual mode. Is it ok to have here 0 as default fallback for empty state?